cardano-api/src/Cardano/Api/Typed.hs

```diff
@@ -1626,69 +1625,63 @@ instance ToJSON MultiSigScript where
            ]
 
 instance FromJSON MultiSigScript where
-  parseJSON = Aeson.withObject "MultiSigScript" $ \obj -> parseMultiSigScript obj
+  parseJSON v = any' v <|> all' v <|> atLeast v
```

Comment:
Oh wait, we're missing `sig` here. The top level is no different to nested layers.

Which means we can simplify by reusing `convertValueToMultiSigScript`.

We can also perhaps make the naming more regular here, e.g.
```
instance FromJSON MultiSigScript where
  parseJSON = parseScript

parseScript :: ...
parseScriptAny :: ... 
parseScriptAll :: ... 
parseScriptAtLeast :: ... 
parseScriptSig :: ... 
```